### PR TITLE
LPD-37074 Define the correct companyId when executing the data provider

### PR DESCRIPTION
**Pull request by vngoliveira**
[LPD-37074](https://liferay.atlassian.net/browse/LPD-37074)

### CODE REVIEW

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/src/main/java/com/liferay/dynamic/data/mapping/data/provider/internal/DDMDataProviderInvokerImpl.java [hunk 1]
```diff
@@ -153,7 +154,7 @@ protected DDMDataProviderResponse doInvokeExternal(
 					DDMRESTDataProviderSettings.class));
 
 		DDMDataProviderResponse ddmDataProviderResponse =
-			ddmDataProviderInvokeCommand.execute();
+			ddmDataProviderInvokeCommand.run();
```
Comment (carolmariaabb) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/src/main/java/com/liferay/dynamic/data/mapping/data/provider/internal/DDMDataProviderInvokerImpl.java:157]: @vngoliveira the code looks good, but could you explain to me why using the `run` method fixes the issue?

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/src/main/java/com/liferay/dynamic/data/mapping/data/provider/internal/DDMDataProviderInvokerImpl.java (hunk 1 above)
Comment (vngoliveira) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/src/main/java/com/liferay/dynamic/data/mapping/data/provider/internal/DDMDataProviderInvokerImpl.java:157]: Hey @carolmariaabb
From what I understand, to work correctly in a new instance, we need to set up the Hystrix thread to connect to the new instance. When you call the run method, it filters the instance in the getData method and gets the data.

Comment (carolmariaabb) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/src/main/java/com/liferay/dynamic/data/mapping/data/provider/internal/DDMDataProviderInvokerImpl.java:157]: I'm a little worried about this change. I was reading the main differences between these methods and here is what I found:

- Use execute() when you want to perform the command's action asynchronously without blocking the current thread. This is useful for long-running or I/O-bound operations.

- Use run() when you need the result of the command immediately and are willing to block the current thread. This is suitable for short-running operations where blocking is acceptable.

Therefore, with your changes, we would change the current behavior of the Data Provider.

- What error is shown when trying to reproduce the issue without the fix?
- Did you try to reproduce the same issue without the `include-and-override=portal-liferay-online.properties in portal-ext.properties` property?

**Comment by vngoliveira** [comment]
ci:test:sf

**Comment by vngoliveira** [comment]
ci:test:relevant

**Comment by vngoliveira** [comment]
ci:test:forms

**Review by carolmariaabb — APPROVED** [review]

**Comment by vngoliveira** [comment]
I tested some scenarios to verify if all will continue working well after applying the fix: 

1. with include-and-override=portal-liferay-online.properties and with database.partition.enabled=true ⇾ it works on the default instance and the new instance also works 
2. with include-and-override=portal-liferay-online.properties and with database.partition.enabled=false ⇾ it works on the default instance and the new instance it also works  
3. without include-and-override=portal-liferay-online.properties and with database.partition.enabled=true ⇾ it works on the default instance and the new instance it also works  
4. without include-and-override=portal-liferay-online.properties and with database.partition.enabled=false ⇾ it works on the default instance and the new instance it also works

**Review by rebsilva — APPROVED** [review]
LGTM! :tada:

**Comment by rebsilva** [comment]
ci:test:relevant

**Comment by rebsilva** [comment]
ci:test:playwright

**Comment by vngoliveira** [comment]
Hey @rebsilva and @carolmariaabb,
I checked the CI results and the failures seem not to be related. Can we forward it?

**Comment by carolmariaabb** [comment]
@vngoliveira do we really need this [commit](https://github.com/liferay-forms/liferay-portal/pull/2617/commits/b9705fd1e78f11f2a1d489f157701bef98a2e919)?

**Comment by vngoliveira** [comment]
> @vngoliveira do we really need this [commit](https://github.com/liferay-forms/liferay-portal/pull/2617/commits/b9705fd1e78f11f2a1d489f157701bef98a2e919)?

Yes, to create the form lar that will be imported into the test

**Comment by carolmariaabb** [comment]
@vngoliveira is this common for playwright tests?

**Comment by carolmariaabb** [comment]
ci:forward

**Comment by rebsilva** [comment]
> @vngoliveira is this common for playwright tests?

i asked @larissacribeiro about this because i reviewed a similar PR before that had this, and she explained to me that  she saw other cases like that already on the code of other teams since 5 months ago, so it's something that is already in use... but we don't know exactly if it's the best approach, so we could ask our TLs about it if you think its necessary @carolmariaabb

**Comment by larissacribeiro** [comment]
Hi, @carolmariaabb! Yes, it is, for both playwright and poshi. This is because the format prevents us from committing an compressed file, so in this commit we are sending the content of a imported .portlet.lar file from forms, but decompressed. That's why on formsPage importFile method, we are [ziping](https://github.com/liferay/liferay-portal/blob/72260c9f00fdd54729e350dc4a62c1338478984f/modules/test/playwright/pages/dynamic-data-mapping-form-web/FormsPage.ts#L146C3-L146C58) the folder back. You can see another example here [export-import-web](https://github.com/liferay/liferay-portal/tree/72260c9f00fdd54729e350dc4a62c1338478984f/modules/test/playwright/tests/export-import-web/dependencies/folder.portlet.lar). Let me know if I can assist with anything else, thanks!

**Comment by carolmariaabb** [comment]
ci:test:stable

**Comment by carolmariaabb** [comment]
ci:test:stable

**Comment by vngoliveira** [comment]
ci:test:stable

**Comment by carolmariaabb** [comment]
ci:test:stable

**Comment by carolmariaabb** [comment]
@vngoliveira the failure is related:

```
junit.framework.AssertionFailedError: Redundant dependency detected in modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/build.gradle expected:<compileOnly project(":core:petra:petra-lang")> but was:<testImplementation project(":core:petra:petra-lang")>
```

**Comment by vngoliveira** [comment]
ci:test:sf

**Comment by vngoliveira** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:forward

**Pull request closed without merging**